File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/model/view/FloatSliderSetting.java
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.features.settings.model.view;

import android.content.Context;

import org.dolphinemu.dolphinemu.features.settings.model.AbstractFloatSetting;
import org.dolphinemu.dolphinemu.features.settings.model.AbstractSetting;
import org.dolphinemu.dolphinemu.features.settings.model.Settings;

public class FloatSliderSetting extends SliderSetting
{
  protected AbstractFloatSetting mSetting;

  public FloatSliderSetting(Context context, AbstractFloatSetting setting, int titleId,
          int descriptionId, int min, int max, String units, int stepSize)
  {
    super(context, titleId, descriptionId, min, max, units, stepSize);
    mSetting = setting;
  }

  public FloatSliderSetting(AbstractFloatSetting setting, CharSequence name,
          CharSequence description, int min, int max, String units)
  {
    super(name, description, min, max, units);
    mSetting = setting;
  }

  public int getSelectedValue()
  {
    return Math.round(mSetting.getFloat());
  }

  public void setSelectedValue(Settings settings, float selection)
  {
    mSetting.setFloat(settings, selection);
  }

  @Override
  public AbstractSetting getSetting()
  {
    return mSetting;
  }
}

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/model/view/FloatSliderSetting.kt
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.features.settings.model.view

import android.content.Context
import org.dolphinemu.dolphinemu.features.settings.model.AbstractFloatSetting
import org.dolphinemu.dolphinemu.features.settings.model.AbstractSetting
import org.dolphinemu.dolphinemu.features.settings.model.Settings
import java.math.BigDecimal
import java.math.MathContext

open class FloatSliderSetting : SliderSetting {
    protected val floatSetting: AbstractFloatSetting

    val min: Float
    val max: Float
    val stepSize: Float

    override val setting: AbstractSetting
        get() = floatSetting

    constructor(
        context: Context,
        setting: AbstractFloatSetting,
        titleId: Int,
        descriptionId: Int,
        min: Float,
        max: Float,
        units: String,
        stepSize: Float,
        showDecimal: Boolean
    ) : super(context, titleId, descriptionId, units, showDecimal) {
        floatSetting = setting
        this.min = min
        this.max = max
        this.stepSize = stepSize
    }

    constructor(
        setting: AbstractFloatSetting,
        name: CharSequence,
        description: CharSequence,
        min: Float,
        max: Float,
        units: String,
        stepSize: Float,
        showDecimal: Boolean
    ) : super(name, description, units, showDecimal) {
        floatSetting = setting
        this.min = min
        this.max = max
        this.stepSize = stepSize
    }

    open val selectedValue: Float
        get() = floatSetting.float

    open fun setSelectedValue(settings: Settings, selection: Float) {
        floatSetting.setFloat(settings, selection)
    }
}
